### ISSUE TITLE
Ensure gauges can only be created for assets that exist on-chain

### ISSUE THREAD

**Issue by ValarDragon** [osmosis-labs/osmosis#530]
Currently gauges seem to be creatable for assets that don't exist on-chain. This doesn't really pose any efficiency problems, but we should nonetheless eliminate this. (If nothing else, just for integrators sanity lol)

Here is an instance for a gauge for the denom "Stake". 
```
    {
      "id": "244",
      "is_perpetual": false,
      "distribute_to": {
        "lock_query_type": "ByDuration",
        "denom": "stake",
        "duration": "604800s",
        "timestamp": "2021-03-16T17:57:55Z"
      },
      "coins": [
        {
          "denom": "ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2",
          "amount": "1000"
        }
      ],
      "start_time": "2021-07-01T20:36:45Z",
      "num_epochs_paid_over": "5",
      "filled_epochs": "0",
      "distributed_coins": []
    },
```

This can be fixed by making the CreateGauge logic in `x/incentives` check that the denomination exists in the bank keeper on the chain, somewhere.

**Comment by jeebster** [comment]
@ValarDragon I've been working on this but am a bit uncertain about a few things with regards to denomination existence logic:

1. Should the check be performed against the [lockup.QueryCondition `Denom`](https://pkg.go.dev/github.com/osmosis-labs/osmosis/x/lockup/types#QueryCondition) or [sdk.Coin `Denom`](https://pkg.go.dev/github.com/cosmos/cosmos-sdk@v0.42.9/types#Coin)? I would think the latter but you mention the `lockup.QueryCondition.Denom` ("stake") in the issue description.

2. Should [supply](https://pkg.go.dev/github.com/cosmos/cosmos-sdk@v0.42.9/x/bank/keeper#BaseKeeper.GetSupply) or [metadata](https://pkg.go.dev/github.com/cosmos/cosmos-sdk@v0.42.9/x/bank/keeper#BaseKeeper.GetDenomMetaData) be referenced for validating asset existence on-chain? I noticed there will be a [HasSupply()](https://github.com/cosmos/cosmos-sdk/issues/10387) method available so I would think `supply` is the standard reference for checking existence, however, technically an asset can exist on chain with no supply (after supply runs out). This seems like an edge case since supplies shouldn't run out for quite some time, but I figured I'd defer to you on the preferred methodology to verify asset existence

Either way, I'm seeing lots of test failures checking both [lockup.QueryCondition `Denom`](https://pkg.go.dev/github.com/osmosis-labs/osmosis/x/lockup/types#QueryCondition) or [sdk.Coin `Denom`](https://pkg.go.dev/github.com/cosmos/cosmos-sdk@v0.42.9/types#Coin)? so it would be nice to have some clarity on the preferred approach

**Comment by ValarDragon** [comment]
The check should basically be:

`bankkeeper.HasSupply(lockup.QueryCondition.Denom)`

We want to upgrade to SDK v0.44 ASAP, so probs best to wait on that. (Or can build it in a separate branch using Jacob's PR)

**Comment by jeebster** [comment]
Sounds good; will wait until this is available via SDK version upgrade

I haven't taken a deep look into the SDK supply code, but I'm making an assumption here that an asset *is not* removed if the supply ultimately reaches zero - it's quite possible I'm incorrect about this expectation.
